extensions/WSLExtension/Strings/en-US/Resources.resw

```diff
@@ -126,28 +126,92 @@
+    <value>Starting the creation process for {0}</value>
+    <comment>{Locked="{0}"} Text for when a user starts the creation process for a specific wsl distribution. {0} is the name of the distribution</comment>
   </data>
   <data name="WSLWaitingToCompleteInstallation" xml:space="preserve">
     <value>Waiting for {0}'s installation to complete</value>
     <comment>{Locked="{0}"} Text message to display when a wsl distribution is installing. {0} is the name of the distribution</comment>
   </data>
+  <data name="WSLWaitingToCompleteRegistration" xml:space="preserve">
+    <value>Waiting for the Windows Subsystem for Linux service to register {0}. This may take a few minutes.</value>
+    <comment>{Locked="Windows", "Linux", "{0}"} Text to display when a specific wsl distribution is being registered. {0} is the name of the distribution</comment>
+  </data>
   <data name="WSLInstallationFailedWithException" xml:space="preserve">
     <value>Unable to install and register {0} due to error: {1}. Try using wsl.exe to install it manually. See aka.ms/wslinstall</value>
-    <comment>{Locked="{0}", "{1}", "{wsl.exe}", {aka.ms/wslinstall}} Text message to display when we failed to install a wsl distribution. {0} is the name of the distribution and {1} is the error message.</comment>
+    <comment>{Locked="{0}", "{1}", "wsl.exe", "aka.ms/wslinstall"} Text message to display when we failed to install a wsl distribution. {0} is the name of the distribution and {1} is the error message.</comment>
   </data>
-  <data name="WSLInstallationFailedTimeOut" xml:space="preserve">
+ <data name="WSLInstallationFailedTimeOut" xml:space="preserve">
     <value>Installation timed out, unable to install and register {0}. Try using wsl.exe to install manually. See aka.ms/wslinstall</value>
     <comment>{Locked="{0}", "{wsl.exe}", "{aka.ms/wslinstall}"} Text message to display when we failed to install a wsl distribution. {0} is the name of the distribution</comment>
   </data>
+  <data name="InstallingWslKernelPackage" xml:space="preserve">
+    <value>The Windows Subsystem for Linux kernel package is not installed. Starting installation.</value>
+    <comment>{Locked="Windows", "Linux"}</comment>
+  </data>
+  <data name="WslKernelPackageInstallationCheck" xml:space="preserve">
+    <value>Checking if the Windows Subsystem for Linux kernel package is installed</value>
+    <comment>{Locked="Windows", "Linux"}</comment>
+  </data>
   <data name="WSLInstallationCompletedSuccessfully" xml:space="preserve">
-    <value>Successfully installed and registered {0}</value>
-    <comment>{Locked="{0}"} Text message to display when we successfully installed a wsl distribution. {0} is the name of the distribution</comment>
+    <value>Successfully installed {0}</value>
+    <comment>{Locked="{0}"} Text to display when we have successfully installed a specific wsl distribution package onto a users machine. {0} is the name of the distribution</comment>

```

Comment:
NIT: Avoid personal pronouns. Consider _Text to display when a specific wsl distribution package is successfully installed onto a user's machine_

Reply:
Hmm, but these are comments to our localization team not the user, but sure I can update them if you want.